### DISCUSSION TITLE
HeatmapLayer dosen't aggregate as expect in high precision

### DISCUSSION THREAD

**Discussion by timeless15** [visgl/deck.gl#5617] [Q&A] [answered]
When I use HeatmapLayer, it dosen't aggregate as expect. I use the ScatterplotLayer to show the original data.
Obviously, when zoom in the map, the scatter points don't aligned to heatmap points. The heatmap points seem aligned vertically, and the layer is shaking when zooming.

![image](https://user-images.githubusercontent.com/25543532/112930583-28ec0d80-914d-11eb-93ae-0ccb90c466e1.png)

But if I remove the integer part of the coordinates, like `[120.15948104329973, 30.27054640617553]`  to `[0.15948104329973, 0.27054640617553]`, it shows as expect:
 
![image](https://user-images.githubusercontent.com/25543532/112931100-132b1800-914e-11eb-946f-a5f6c020d4d6.png)

[Here is the code](https://codesandbox.io/s/jovial-moon-nveow?file=/src/index.js), you can click the toggle button to see different data behaviors.

I'm using Deck.GL for indoor map, which means the coordinate has more digits, and zoom level is set above 18. So I guess it's due to precision loss, I tried the `fp64` prop, but it didn't work. I still can't figure out, can anyone help me? Thanks a lot🙏

**Comment by Pessimistress** [comment, marked as answer]
The `fp64` prop was deprecated back in v6. Support for high-precision seems to be a bug.

**Reply by timeless15** [reply]
But I tried `Fp64Extension`, didn't work either.
Do you mean the mechanism of supporting high precision in Heatmap still has bug?

**Reply by timeless15** [reply]
Oh I see the [pull request](https://github.com/visgl/deck.gl/pull/5621?w=1).
Thanks a lot 😊 👍
